perf/config.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2020 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Configuration for perf event manager.
+package perf
+
+import (
+	"encoding/json"
+	"fmt"
+	"k8s.io/klog"
+	"strconv"
+)
+
+type Events struct {
+	// List of perf events' names to be measured. Any value found in
+	// output of perf list can be used.
+	Events [][]Event `json:"events"`
+
+	// List of custom perf events' to be measured. It is impossible to
+	// specify some events using their names and in such case you have
+	// to provide lower level configuration.
+	CustomEvents [][]CustomEvent `json:"custom_events"`
```

Comment:
I was asking a slightly different question.  Can "simple" and "custom" events be grouped together?  My thinking was that we could have part of the config for defining all of the events, and another for specifying their groupings.  E.g.

```json
{
  "events": [
    ["instructions", "instructions_retired"]
  ],
  "custom_events":   [
    {
      "type": 4,
      "config": [
        "0x5300c0"
      ],
      "name": "instructions_retired"
    }
  ]
}
```

Reply:
Oh, ok. I didn't get this. I bet it is doable :)